Examples/Io/HepMC3/src/HepMC3Writer.cpp

```diff
@@ -90,19 +97,29 @@ std::unique_ptr<HepMC3::Writer> HepMC3Writer::createWriter(
 
 HepMC3Writer::~HepMC3Writer() = default;
 
+ProcessCode HepMC3Writer::beginEvent(std::size_t threadId) {
+  ACTS_VERBOSE("Begin event, next_event=" << m_nextEvent);
+  if (!m_cfg.writeEventsInOrder) {
+    // Nothing to do if we don't write in order
+    return ProcessCode::SUCCESS;
+  }
+
+  ACTS_DEBUG("thread=" << threadId << ", next_event=" << m_nextEvent
+                       << " waiting for semaphore");
+  m_waiting++;
+  m_queueSemaphore.try_acquire_for(std::chrono::seconds(10));
```

Comment:
Hmm, why don't we use the return value from this? If the semaphore fails to acquire, we just fail silently? Why not simply block until it succeeds?

Reply:
I'm not sure I understand. 

- What return value to you mean?
- If I use just `acquire()` it will block forever, so if there's a bug somewhere, I'm stalling the whole program. That's why I'm hesitant to wait without a timeout.
- If the semaphore fails to acquire and we fail silently, that would mean we silently skip processing that event. That seems confusing to me

Reply:
Sorry I meant the return value of `try_acquire_for`. That returns a boolean indicating the success for failure of the lock; I'm not sure if it makes sense to call `try_acquire_for` without checking if it succeeded.